File: app/src/test/java/ru/androidschool/rxjava/CreateObservable.java
package ru.androidschool.rxjava;

import org.junit.Test;
import java.util.concurrent.Callable;
import io.reactivex.rxjava3.core.Observable;

public class CreateObservable {

    @Test
    public void justDemoSample() {
        Observable<String> source = Observable.just("Hello");
        source.subscribe(s -> System.out.println("RECEIVED: " + s));
    }

    @Test
    public void fromCallableDemoSample() {

        Observable<String> source =
                Observable.fromCallable(new Callable<String>() {
                    @Override
                    public String call() throws Exception {
                        return Long.toString(System.currentTimeMillis());
                    }
                });

        source.subscribe(s -> System.out.println("RECEIVED: " + s));
    }

}

File: app/src/test/java/ru/androidschool/rxjava/kotlin/create/CreateObservable.kt
package ru.androidschool.rxjava.kotlin.create

import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single
import org.junit.Test


class CreateObservable {

    @Test
    fun justDemoSample() {
        val source = Observable.just("Hello")
        source.subscribe { s -> println("RECEIVED: $s") }
    }

    @Test
    fun fromCallableDemoSample() {
        val source = Observable.fromCallable { System.currentTimeMillis().toString() }
        source.subscribe { s -> println("RECEIVED: $s") }
    }

    @Test
    fun justSingleDemoSample() {
        val source = Single.just("Hello")
        source.subscribe { s -> println("RECEIVED: $s") }
    }

}
